public/app/features/dashboard/panel_editor/QueriesTab.tsx

```diff
@@ -195,8 +186,9 @@ export class QueriesTab extends PureComponent<Props, State> {
   };
 
   onAddMixedQuery = (datasource: any) => {
-    this.onAddQuery({ datasource: datasource.name });
+    this.props.panel.targets = addQuery(this.props.panel.targets, { datasource: datasource.name });
     this.setState({ isAddingMixed: false, scrollTop: this.state.scrollTop + 10000 });
```

Comment:
Maybe we could extract this part `this.state.scrollTop + 10000` in a function instead and call that?